assets/asset_manager.cc

```diff
@@ -29,6 +29,16 @@ void AssetManager::PushBack(std::unique_ptr<AssetResolver> resolver) {
   resolvers_.push_back(std::move(resolver));
 }
 
+void AssetManager::Merge(std::shared_ptr<AssetManager> assetManager) {
```

Comment:
About the name, `Merge` typically means that all the elements will be combined somehow, but it doesn't look like that's what this is doing.

I'd like to suggest a better name, but I think I don't understand what is special about the last resolver of `assetManager`.

Reply:
Maybe `PreserveBundleManager` or something like that?

The very last resolve is the first one ever added, which is the one that resolves assets from the APK/IPA bundle that is installed with the device. Preserving it means that we don't need to sync every asset into the devFS when connecting to a device

Reply:
Would something like this work?
* `AssetManager` adds an API to declare which asset resolvers should be preserved across a hot reload
* `Shell::OnServiceProtocolSetAssetBundlePath` retains those resolvers when constructing the new `AssetManager`

Reply:
I think it would have to be done in engine.cc since the current asset manager will be private. I could add a param to the directory asset resolver that controls whether it should be preserved and then set this to true for the initial resolvers.